### PR TITLE
chore: Add more build artifacts for debugging

### PR DESCRIPTION
**Pull request by lorensr**
- add npm logs for https://github.com/temporalio/sdk-typescript/issues/496#issuecomment-1166099476
- always upload nightly worker logs, so we have green output to compare to

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -63,6 +63,14 @@ jobs:
           # TODO: Run integration tests on MacOS / Windows probably using temporalite
           RUN_INTEGRATION_TESTS: ${{ startsWith(matrix.os, 'ubuntu') }}
 
+      - uses: actions/upload-artifact@v3
```
Comment (bergundy) [inline .github/workflows/ci.yml:66]: What's this for?

File: .github/workflows/stress.yml [hunk 2]
```diff
@@ -54,11 +54,10 @@ jobs:
       - run: mkdir -p $TEMPORAL_TESTING_LOG_DIR/tails
       - run: mkdir -p $TEMPORAL_TESTING_MEM_LOG_DIR
       - run: 'timeout ${{ inputs.test-timeout-minutes }}m npm run ${{ inputs.test-type }}'
-      # Tail the logs and upload as artifact on cancel or failure
       - run: for f in $TEMPORAL_TESTING_LOG_DIR/*.log; do tail -20000 $f > $TEMPORAL_TESTING_LOG_DIR/tails/$(basename $f); done
-        if: ${{ cancelled() || failure() }}
+        if: ${{ always() }}
       - uses: actions/upload-artifact@v3
-        if: ${{ cancelled() || failure() }}
+        if: ${{ always() }}
```
Comment (bergundy) [inline .github/workflows/stress.yml:60]: Assuming this is temporary

**Comment by bergundy** [comment]
Not required anymore, we found the nightly issue

**Comment by bergundy** [comment]
Re-opened, sorry, I missed the first point in the description

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -63,6 +63,14 @@ jobs:
...
+        if: ${{ cancelled() || failure() }}
```
Comment (bergundy) [inline .github/workflows/ci.yml:67]: Can do only on failure

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bergundy) [reply .github/workflows/ci.yml:66]: If I'd read the description I'd have known

File: .github/workflows/stress.yml (hunk 2 above)
Comment (lorensr) [reply .github/workflows/stress.yml:60]: This is to have green output to compare to. Can compare to local output, but may be different due to platform or length?

Comment (bergundy) [reply .github/workflows/stress.yml:60]: Should be similar to local, I guess it wouldn't hurt keeping

**Review by bergundy — APPROVED** [review]

**Pull request merged**